Issue1193 improve docs for ve df (#1194)
oceanprotocol/docs: veocean-data-farming/df-background.md

---
description: Intuition and background on the Data Farming program.
---
# Data Farming Background

> 

Reviewer: Add a line at the top to summarize the goal:

"Data Farming (DF) incentivizes for growth of Data Consume Volume (DCV) in the Ocean ecosystem."

Author: Done

> It rewards OCEAN to liquidity providers (stakers) as a function of consume volume and liquidity. It’s like DeFi liquidity mining, but tuned for data consumption. DF’s aim is to achieve a minimum supply of data for network effects to kick in, and once the network flywheel is spinning, to increase growth rate.

Reviewer 2: It might be clearer just to describe them as stakers. People might assume that we're running an AMM if we use terms like liquidity providers

Reviewer: Yes, agree. We shouldn't use "liquidity provider" anywhere, it's inaccurate.

Author: Yeah, I agree. Done.

## Active Work to Drive APY

Data Farming is not a wholly passive activity. The name of the game is to drive Data Consume Volume (DCV). High APYs happen only when there is sufficiently high DCV. High DCV means publishing and consuming truly useful datasets (or algorithms).

Thus, if you really want to max out your APY:
- create & publish datasets (and make $ in selling them) — or work with people who can

> - consume the datasets (to make $) — or work with people who can

Reviewer 2: "consume the datasets" sounds a bit like wash consume. I'd prefer if this said something like:

```suggestion
- Promote the consumption of the datasets — or work with people who can
```

Reviewer: "promote the consumption" is too watered-down. 

And we want to emphasize making $.

How about "buy the datasets and use them to make $ - or work with people who can"

Author: Fixed CTA. Kept formatting consistent with the rest, made sentences uppercase.

```
Thus, if you really want to max out your APY:
- Create & publish datasets (and make $ in selling them) — or work with people who can
- Buy the datasets (and use them to make $) — or work with people who can
- Stake on them, and claim the rewards.
```

> - go stake on them, and finally claim the rewards.

Reviewer 2: Logically it would make sense if this was the second item on the list

Author: ```
if this was the second item on the list
```
If "this" what?

Based on your feedback, I can't make out what you're recommending to me. What's the status quo, what should it be changed into?

Can you number the items and tell me the order you want them in?

Author: @jamiehewitt15 can we resolve this?

Reviewer 2: I was referring to "go stake on them", my suggestion is that it should come before "consume the datasets (to make $) — or work with people who can". So the list looks something like this: 

1. create & publish datasets (and make $ in selling them) — or work with people who can
2. go stake on them
3. consume the datasets (to make $) — or work with people who can
4. finally claim the rewards.

Author: I failed to capture this although this is how other tutorials are setup.
I'll update right away.

Author: Resolved

> Driving DCV for publishing & consuming is your challenge. It will take real work. And then the reward is APY. It’s incentives all the way down:)

Author: Done

## Reward Function

The Reward Function (RF) governs how active rewards are allocated to stakers.

Rewards are calculated as follows:

> 1. Distribute OCEAN across each asset based on rank: highest-DCV asset gets most OCEAN, etc.
> 1. For each asset and each veOCEAN holder:

Reviewer 2: ```suggestion
- Distribute OCEAN across each asset based on rank: highest-DCV asset gets most OCEAN, etc.
- For each asset and each veOCEAN holder:
```

Author: Your feedback is incorrect.
The correct way of creating a numbered list in markdown is.
```markdown
1. One
1. Two
1. Three
```

– If the holder is a publisher, 2x the effective stake
– Baseline rewards = (% stake in asset) * (OCEAN for asset)
– Bound rewards to the asset by 125% APY
– Bound rewards by asset’s DCV * 0.1%. This prevents wash consume.

You can find this code inside [calcrewards.py](https://github.com/oceanprotocol/df-py/blob/main/util/calcrewards.py) in the Ocean Protocol [df-py repo](https://github.com/oceanprotocol/df-py/)

## Data Assets that Qualify for DF

Data assets that have veOCEAN allocated towards them get DF rewards.

> The data asset may be of any type — dataset (for static URIs) or algorithm for Compute-to-Data. The data asset may be fixed price or free price. If fixed price, any token of exchange is alright (OCEAN, H2O, USDC, ..).

Reviewer 2: Are you sure we support data farming on free assets? 
- How do we rank consume volume of free assets? DCV will always be $0
- Also, if you can data farm on free assets, wash consume is still possible.

Author: They are!
I have updated the **Pricing FAQ** section to include
```
<details>
<summary>In this scheme, can people stake on free datasets?</summary>

Yes. They allocate their veOCEAN to datasets. Then DF rewards follow the usual DF formula: DCV * veOCEAN stake. Except in this case although DCV is 0, the gas fees will still count towards calculating rewards.
</details>
```

To qualify for DF, a data asset must also:
- Have been created by Ocean Smart contracts [deployed](https://github.com/oceanprotocol/contracts/blob/v4main/addresses/address.json) by OPF to [production networks](https://docs.oceanprotocol.com/core-concepts/networks)
- Be visible on [Ocean Market](https://market.oceanprotocol.com/)
- Can’t be in [purgatory](https://github.com/oceanprotocol/list-purgatory/blob/main/policies/README.md)

## 3 Phases of Data Farming

> Data Farming has evolved over time and will continue to do so as the Emission Curve progresses. Below are important dates and parameters incurred during the evolution of the Data Farming program.

Reviewer 2: They're actually not important dates because they are in the past. I'd suggest the following: 

```suggestion
Data Farming has evolved over time and will continue to do so as the Emission Curve progresses. We are now in DF main, below are the previous phases and parameters incurred during the evolution of the Data Farming program.
```

Author: Done!

**DF Alpha** - Counting starts Thu June 16. 10K OCEAN rewards are budgeted per week. Rewards are distributed at the end of every week, for the activity of the previous week. It runs 4 weeks. The aim is to test technology, learn, and onboard data publishers.

> **DF Beta** - Counting starts Thu July 14. Rewards are up to 100K OCEAN per week. It runs up to 20 weeks. The aim is to test the effect of larger incentives, learn, and refine the technology.

Reviewer 2: You've written this in the present tense which is confusing as these are in the past. It's made worse because you haven't included the year, so next January people will be very confused. The phases that have ended should say something like 

"Ended on {Date}. 10K OCEAN rewards were budgeted per week. Rewards were distributed at the end of every week, for the activity of the previous week. It ran for 4 weeks. The aim was to test technology, learn, and onboard data publishers."

Author: Done. Please (see update and comments above)[https://github.com/oceanprotocol/docs/pull/1194#discussion_r1148250617].

> **DF Main** - Immediately follows DF Beta. Rewards are up to 718K OCEAN per week. DF Main emits 503.4M OCEAN worth of rewards and lasts for decades. Expected APY is 125% over many months (once fully ramped), staying generous over the long term.

Reviewer: Incorrect. There are four phases. See the "DF Main" blog post which lays this out.

Author: I do see this elsewhere now...

Perhaps here in section 4, the subsections should be structured such that they follow Phase 1 => 4.
```4. DF Phases & Evolution```
4.1..4.4 - DF Main Round 29+
4.5 - DF Evolution
https://blog.oceanprotocol.com/ocean-data-farming-is-launching-f633532d831c

I believe I took this from an existing blog post, so I will try to find the exact reference and post it here so we can resolve this.

Author: I have pushed an update to communicate all 4 phases.

However, I have re-worded everything to use past-tense, as per feedback from @jamiehewitt15 in the link here.

Reviewer: You reference the post for DF1.

Obviously DF evolves. The "status" of DF = the initial post plus the "diff" from every post after that.

We're not going to go and change every single post of the past. Rather, they serve as a historical record of the evolution of DF.

So that it's easier for people to know how DF works "now", now and then I write a post that describes how DF works, end-to-end.
- I did this when we launched veOCEAN & re-launched DF. This was the beginning of DF/VE alpha
- And I did this last week when we launched DF Main.

Therefore the "golden truth" for how DF works is in the "DF main" post at time of DF29. _NOT_ in the "DF is launching post of DF1.

Author: ```
Obviously DF evolves. The "status" of DF = the initial post plus the "diff" from every post after that.

We're not going to go and change every single post of the past. Rather, they serve as a historical record of the evolution of DF.

So that it's easier for people to know how DF works "now", now and then I write a post that describes how DF works, end-to-end.

I did this when we launched veOCEAN & re-launched DF. This was the beginning of DF/VE alpha
And I did this last week when we launched DF Main.
Therefore the "golden truth" for how DF works is in the "DF main" post at time of DF29. NOT in the "DF is launching post of DF1.
```
Sorry, I should have realized this.
I thought I had pointed somewhere further along.
I agree.

I hope the update in place is good.
